framework/src/utils/PetscDMMoose.C

```diff
@@ -743,6 +743,19 @@ DMMooseGetEmbedding_Private(DM dm, IS * embedding)
               if (edof >= dofmap.first_dof() && edof < dofmap.end_dof())
                 indices.insert(edof);
           }
+          // loop over lower dimensional slave elements
+          for (const auto & bit : *(dmm->_block_names))
+            if (bit.second.find("_slave_subdomain") != std::string::npos)
```

Comment:
The subdomain name should not be hard-coded.  @lindsayad  Did you use this name by default when you creating a subdomain for mortar?

Reply:
This `DMMoose` stuff is hard-coded to work with a `Contact` block. And yes in `Contact` we automatically set up the lower-dimensional subdomains with these guaranteed subdomains. I actually voiced this exact same concern to @dewenyushu, but she correctly pointed out this fact to me. If we generalize `DMMoose` to work with arbitrary constraints then we won't be able to do what she's doing here, and we'll have to add some kind of map to `MooseMesh` for retrieving lower-d subdomains associated with sidesets. But for now I think this is ok.

Reply:
Yes, It may be the time to refactor DMMoose to take arbitrary blocks. The basic idea is simple here: we just "abuse" FS to mimic block elimination.  

It would be helpful if you @lindsayad could add API in MooseMesh so that we could retrieve this info easily. Hard-coded thing is always bad.

We could leave this PR as is, but we should make these changes in the future

Reply:
Yes, I think that that is beyond @dewenyushu's scope here. I can add that `MooseMesh` support after this goes in, and use her tests as a bellwether for that new support

Reply:
Great